### ISSUE TITLE
Twitter sources

### ISSUE THREAD

**Issue by MizouziE** [Energy-Price-News-API/energy-prices-api#3]
Add a twitter scraper using customizable search filters![image](https://user-images.githubusercontent.com/90829439/162581284-7bb2fd98-36a0-4a67-a572-024596861221.png)

**Comment by MizouziE** [comment]
After attempting this, I've learnt that there is a twitter public API already. This needs to be put to use instead and I'm open to discussion on what would be a good implementation.

**Comment by TAKANOME-DEV** [comment]
Ah ok, so you want to use that twitter API to retrieve some tweets ?? Is that right ??

**Comment by MizouziE** [comment]
Yes I think it would be better than the way I was doing it by concatenation to the query string. 
I need to read the docs and see what would be the best way. Always open to suggestions though!

**Comment by TAKANOME-DEV** [comment]
Ok cool, I also have to check the docs.
Let me know if you find something interesting!

**Comment by leonlafa** [comment]
I'm happy to take this one on if you can provide me with specifications on how you want users to use this resource 😄

**Comment by MizouziE** [comment]
Thanks @leonlafa 
I thought it should be similar to how the article scraper works by running a search on each twitter user account for tweets containing each of the keywords. 

The difficult part will be figuring out how to basically make calls to the twitter api so many times without breaking fair use rules, hence the need to properly investigate it. So far I have not had the time. 

The desired end result would be a JSON response with the same structure as the news articles response, containing a list of objects each having;
- the retrieved related tweet
- the name of the account
- a url to the tweet
- account profile image (if possible)

It all depends what is possible. So if you have experience with the twitter api or the time to look into it we can discuss how to approach it unless you feel comfortable taking it on? 

Getting something equal to the api/news endpoint would be the most important first step. We can build out the rest later to match something closer to what the api/news/{options} currently look like.

**Comment by leonlafa** [comment]
@MizouziE unfortunately I'm going to have to jump off this one.  I've been extremely busy with work.  I'm unsure when I'll get the change to sit down and get this done.

**Comment by MizouziE** [comment]
@leonlafa No worries bro, I know how it gets! I'll probably work on this very slowly anyway so feel free to chuck any ideas or input this way. Appreciate the heads up 🙏

**Comment by MayurK-cmd** [comment]
Hey can i work on this?

**Comment by MayurK-cmd** [comment]
Hey can i work on this?
